### PR TITLE
Fix NM.Client.new_async parameters

### PR DESCRIPTION
**Pull request by Lihis**
The [new_async](https://gjs-docs.gjsify.org/classes/NM_1_0.NM.Client.html#new_async) requires two parameters (cancellable and callback). As the `new_async` returns void the callback has to be used.

However, I have to admit that I can not guarantee the client usage is now 100% correct.

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -56,14 +57,17 @@ export default class SaneAirplaneMode extends Extension {
...
-                this._client.wireless_enabled = arg;
+                if (this._client !== null)
+                    this._client.wireless_enabled = arg;
             }).bind(this),
 
             getAirplaneModeEnabled: (function() {
-                return this._rfkillManager.airplaneMode;
+                // FIXME: why null is returned? Is the following assumption safe?
+                let ret = this._rfkillManager.airplaneMode;
+                return ret === null ? true : ret;
```
Comment (Lihis) [inline extension.js:70]: On my testing at least the first call landing here got `null` from the `this._rfkillManager.airplaneMode` but I am not sure why.. And I don't know is it safe to assume the airplane mode be enabled when getting `null`~~; but on my machine when I boot the airplane mode seems to be enabled~~.

EDIT: Seems that airplane mode on boot depends.. Normally I have both WiFi and Bluetooth disabled so during testing this I observed the airplane mode to be enabled on boot. Now I enabled WiFi, installed updates, rebooted (with WiFi still enabled, Bluetooth disabled) and the airplane mode was not enabled.

**Comment by AntiKippi** [comment]
First, thank you very much for you contribution (again)!

However this time I don't quite see the benefit in changing the initialization method. Yes using it like this deviates from the documentation, but even the GNOME developers themselves do that (see [here](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/status/network.js#L1998)). So it probably is fine.

Also this NM Client initialization stuff (and this extension in general) is rather fragile. I don't know what the hell is going on there but even changing from asynchronous to synchronous initialization breaks the extension. So I'd rather not risk introducing bugs here.

As for the `this._rfkillManager.airplaneMode` returning `null`, this is rather weird and something one could investigate. The effects of this should be rather limited though, `null` will be implicitly converted to `false` by JavaScript. Since we can not possibly guess the current state of the airplane mode (at least not until #7 is done and even then only if the user has activated the setting we know it to be `false`) `false` seems to be a reasonable fallback for me.

So I am closing this for now. But still thank you for your effort, I really appreciate it.

**Comment by Lihis** [comment]
If the `this._client = await NM.Client.new_async(null);` is used I get this error:

```
Mar 11 06:52:01 gnome-shell[2967]: TypeError: function NM.new_async: At least 2 arguments required, but only 1 passed
                                   
 Stack trace:
   _init@file:///home/tomlah/.local/share/gnome-shell/extensions/sane-airplane-mode@kippi/extension.js:44:40
   enable@file:///home/tomlah/.local/share/gnome-shell/extensions/sane-airplane-mode@kippi/extension.js:89:14
   _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:253:38
   loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:461:32
   async*_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:759:24
   async*_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:765:48
   _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:800:20
   async*init@resource:///org/gnome/shell/ui/extensionSystem.js:72:14
   _initializeUI@resource:///org/gnome/shell/ui/main.js:303:22
   start@resource:///org/gnome/shell/ui/main.js:176:11
   @resource:///org/gnome/shell/ui/init.js:12:47
   @resource:///org/gnome/shell/ui/init.js:21:20
```

but adding a 2nd null was not enough as the function returns void, at least according the documentation I linked and on my testing it was always `null`.

> ..snip..
> but even the GNOME developers themselves do that (see [here](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/status/network.js#L1998)).
> ..snip..

that is interesting :thinking: I think that code works because of [this](https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/5442266f2882d5ddbf1d714103ee0c13d500e399#0b8a19ae4a89a6099582c3282b83dc95694322c7_22_19):

```javascript
Gio._promisify(NM.Client, 'new_async');
```

and it seems to do the trick so I'll open up a new PR.

**Comment by AntiKippi** [comment]
Ha that's interesting, with Arch and GNOME 46 I don't get this error. What OS and GNOME Shell version are you testing this with?

**Comment by Lihis** [comment]
I'm on Fedora 39 with GNOME version 45 (gnome-shell 45.5, gjs 1.78.3, NetworkManager 1.44.2). I don't know would this be v45 specific then.

**Comment by AntiKippi** [comment]
Yes most likely, I am getting the same warning if I boot into a VM with GNOME 45. Never noticed this tho. Thanks for helping out!

**Comment by Lihis** [comment]
No problem and thanks for making the this extension!

**Pull request closed without merging**